tests/framework/web/WebActionTest.php

```
@@ -136,4 +143,122 @@ public function testLegacyConstructorWebActionObservesNullIdInInit(): void
+        $module = new Module('mymod', Yii::$app);
+
+        $module->actionMap = ['guarded' => GuardedAction::class];
+
+        try {
+            $this->expectException(MethodNotAllowedHttpException::class);
+
+            $module->runAction('guarded');
+        } finally {
+            $this->restoreRequestMethod($originalMethod);
+        }
+    }
```

PHPMD (2.15.0)

[error] testThrowMethodNotAllowedHttpExceptionWhenVerbFilterRejectsMethod accesses the super-global variable $_SERVER. (undefined)

(Superglobals)